express-notes-api/index.js

```diff
@@ -0,0 +1,113 @@
+const fs = require('fs');
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+
+app.use(express.json());
+
+app.get('/api/notes', function (req, res) {
+  const notesArray = [];
+  for (const property in data.notes) {
+    notesArray.push(data.notes[property]);
+  }
+  res.status(200).json(notesArray);
+});
+
+app.get('/api/notes/:id', function (req, res) {
+  const id = Number(req.params.id);
+  let outcome = {};
+  outcome.error = `Cannot find note with id ${id}`;
+  let statusCode = 404;
+  if (id < 0) {
+    statusCode = 400;
+    outcome.error = 'id must be a positive integer.';
+  } else {
```

Comment:
While the `400 Bad Request` triggers for negative `id`s, it should also trigger for other values that aren't positive integers (non-numbers, zero, non-whole numbers).